render/egl.c

```diff
@@ -201,32 +213,37 @@ struct wlr_egl *wlr_egl_create_with_drm_fd(int drm_fd) {
 
 	if (eglBindAPI(EGL_OPENGL_ES_API) == EGL_FALSE) {
 		wlr_log(WLR_ERROR, "Failed to bind to the OpenGL ES API");
-		goto error;
+		return NULL;
 	}
 
-	egl->gbm_device = gbm_create_device(drm_fd);
-	if (!egl->gbm_device) {
-		wlr_log(WLR_ERROR, "Failed to create GBM device");
-		goto error;
+	return egl;
+}
+
+static bool egl_init(struct wlr_egl *egl, EGLenum platform,
+		void *remote_display) {
+
+	if (platform == EGL_PLATFORM_GBM_KHR && !egl->exts.KHR_platform_gbm) {
```

Comment:
We should do a similar check for `EGL_PLATFORM_DEVICE_EXT`.

Reply:
I changed the logic to check for `EGL_PLATFORM_DEVICE_EXT` before getting the device from eglQueryDevicesEXT, and a check to `EGL_PLATFORM_GBM_KHR` before creating the GBM device.